src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -346,14 +343,7 @@ private HttpClient getHttpClient(PrintStream logger, AbstractBuild<?, ?> build)
         String stashServer = stashServerBaseUrl;
         DescriptorImpl descriptor = getDescriptor();
 
-		// Determine if we are using the local or global settings
-		String credentialsId = getCredentialsId();
-		if (StringUtils.isBlank(credentialsId)) {
-			credentialsId = descriptor.getCredentialsId();
-		}
-
-		Credentials credentials = CredentialsMatchers.firstOrNull(CredentialsProvider.lookupCredentials(CertificateCredentials.class,
-				Jenkins.getInstance(), ACL.SYSTEM), CredentialsMatchers.withId(credentialsId));
+        CertificateCredentials certificateCredentials = getCredentials(CertificateCredentials.class, build.getProject());
```

Comment:
Wrapping CertificateCredentials with the same logic is invalid as you share the same Credentials ID for REST requests and for CertificateCredentials. They would not work together

If you want to avoid errors with self-signed certs - there should be separate credential id for it.